app/api/settings/settings.ts

```diff
@@ -0,0 +1,200 @@
+import translations from 'api/i18n/translations';
+
+import { Settings, SettingsLinkSchema, SettingsFilterSchema } from 'shared/types/settingsType';
+import { ensure } from 'shared/tsUtils';
+import { LanguageSchema, ObjectIdSchema } from 'shared/types/commonTypes';
+import { validateSettings } from 'shared/types/settingsSchema';
+
+import { settingsModel } from './settingsModel';
+import templates from 'api/templates';
+import { TemplateSchema } from 'shared/types/templateType';
+
+const DEFAULT_MAP_TILER_KEY = 'QiI1BlAJNMmZagsX5qp7';
+
+const getUpdatesAndDeletes = (
+  newValues: (SettingsLinkSchema & SettingsFilterSchema)[] = [],
+  currentValues: (SettingsLinkSchema & SettingsFilterSchema)[] = [],
+  matchProperty: keyof (SettingsLinkSchema & SettingsFilterSchema),
+  propertyName: keyof (SettingsLinkSchema & SettingsFilterSchema)
+) => {
+  const updatedValues: { [k: string]: any } = {};
+  const deletedValues: string[] = [];
+
+  currentValues.forEach((value: SettingsFilterSchema & SettingsLinkSchema) => {
+    const matchValue = newValues.find(
+      v => v[matchProperty] && v[matchProperty]?.toString() === value[matchProperty]?.toString()
+    );
+
+    if (matchValue && matchValue[propertyName] !== value[propertyName]) {
+      if (value[propertyName]) {
```

Comment:
I think this second "if" condition can be combined with the previous one right?

Reply:
you are right, done 👍